client/js/upload.js

```diff
@@ -0,0 +1,53 @@
+"use strict";
+const $ = require("jquery");
+const socket = require("./socket");
+const SocketIOFileUpload = require("socketio-file-upload/client");
+const instance = new SocketIOFileUpload(socket);
+
+instance.listenOnInput(document.getElementById("upload-input"));
+instance.listenOnDrop(document.getElementById("chat-container"));
+
+$("#upload").on("click", () => {
+	$("#upload-input").trigger("click");
+});
+
+instance.addEventListener("complete", function() {
+	// Reset progressbar
+	$("#upload-progressbar").width(0);
+});
+
+instance.addEventListener("progress", function(event) {
+	const percent = `${((event.bytesLoaded / event.file.size) * 100)}%`;
+	$("#upload-progressbar").width(percent);
+});
+
+instance.addEventListener("error", function(event) {
+	// Reset progressbar
+	$("#upload-progressbar").width(0);
+	$("#loading-page-message, #connection-error").addClass("shown");
+	$("#loading-page-message, #connection-error").text(event.message);
+});
+
+/** https://css-tricks.com/drag-and-drop-file-uploading/ **/
+function isDragDropSupported() {
+	const div = document.createElement("div");
+	return (("draggable" in div) || ("ondragstart" in div && "ondrag" in div)) && "FormData" in window && "FileReader" in window;
+}
+
+if (isDragDropSupported()) {
```

Comment:
This check isn't really necessary because you're only binding events in the `if`, and they simply won't do anything if there's no support.

Also the fact that all the evergreen browsers have support for it.